runtime/parachains/src/disputes/slashing.rs

```diff
@@ -568,6 +568,7 @@ impl<T: Config> Pallet<T> {
 	fn initializer_on_new_session(session_index: SessionIndex) {
 		// This should be small, as disputes are limited by spam slots, so no limit is
 		// fine.
+		// TODO: without spam slots is this still valid?
```

Comment:
Without spam slots is `u32::MAX` still a good value?

Reply:
Probably not. However, the number of *concluded* disputes should be limited and this is fine in this case. The commend needs to be updated though.

The bigger problem is in `Disputes::initializer_on_new_session`. There we likely need to support multi-block cleanup (by storing some extra state).

Reply:
nvm

The storage in slashing should only be written to only if there a past session dispute (e.g. a dispute on session boundaries). The storage should be cleaned up in practice once we have #5947, namely here:
https://github.com/paritytech/polkadot/blob/356ca11d1d133f4405d088343850c22878fad69f/runtime/parachains/src/disputes/slashing.rs#L519

Ideally, we should also do multi-block cleanup here. But if we don't, in the worst case some relay blocks on session boundaries will take longer to import unless I'm missing something.